lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1194,6 +1196,23 @@ def embedTTFType42(font, characters, descriptor):
             wObject = self.reserveObject('Type 0 widths')
             toUnicodeMapObject = self.reserveObject('ToUnicode map')
 
+            print(f"SUBSET {filename} characters: "
+                  f"{''.join(chr(c) for c in characters)}")
+            fontdata = self.getSubset(
+                filename,
+                ''.join(chr(c) for c in characters)
+            )
+            print(f'SUBSET {filename} {os.stat(filename).st_size}'
+                  f' ↦ {len(fontdata)}')
+
+            # reload the font object from the subset
+            # (all the necessary data could probably be obtained directly
+            # using fontLib.ttLib)
+            with tempfile.NamedTemporaryFile(suffix='.ttf') as tmp:
+                tmp.write(fontdata)
+                tmp.seek(0, 0)
+                font = FT2Font(tmp.name)
```

Comment:
FT2Font can handle file objects too (things that have a read()), so you don't need to go through the filesystem; a BytesIO should be good enough.

Reply:
I see, this can be done for PDF backend (however, see note on PS backend)